### PR TITLE
OAuth Implementation for Facebook

### PR DESCRIPTION
**Pull request by tochman**
* Adds `redux-oauth` as dependency, and taps into `devise_token_auth` and authorizes user with Facebook credentials
* Uses `validateCredentials()` to make use of `redux_token_auth` (that we use for email/password authentication)

## Screen shot
phadder_oauth_spike

### Tests
Regarding testing OAuth in Cypress: https://docs.cypress.io/guides/references/known-issues.html#OAuth

### CODE REVIEW

**Comment by erikwesterberg** [comment]
NIce work!  and also nice refactoring in other files

File: cypress/integration/visitorCanAuthorizeUsingOAuth.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    cy.visit("http://localhost:3001", {
+      onBeforeLoad(win) {
+        cy.stub(win, 'open')
+      }
+    });
+    cy.get("#login-button").click();
+    cy.get(".button .facebook").click()
+    cy.window().its('open').should('be.called')
+  });
+
```
Comment (pedbr) [inline cypress/integration/visitorCanAuthorizeUsingOAuth.js:13]: shouldn't we have an assertion to test when the facebook auth goes through? and a sad path when it doesn't?

File: src/components/Navbar.jsx [hunk 2]
```diff
@@ -6,9 +6,8 @@ import "../css/style.css";
 
 const Navbar = props => {
   let logInActions;
-  props.currentUser.isSignedIn === false
-    ? (logInActions = <LogIn />)
-    : (logInActions = "");
+  !props.currentUser.isSignedIn
+    && (logInActions = <LogIn />)
```
Comment (pedbr) [inline src/components/Navbar.jsx:10]: this was cool!

**Review by pedbr — COMMENTED** [review]
It would be good if we could go over everything in this PR together since there's a few things that I quite don't fully understand what they are doing. But awesome work!

File: cypress/integration/visitorCanAuthorizeUsingOAuth.js (hunk 1 above)
Comment (tochman) [reply cypress/integration/visitorCanAuthorizeUsingOAuth.js:13]: Check the link in the README

**Review by erikwesterberg — COMMENTED** [review]
I have just commented before, I hadn't pushed the review button. sry for that. But I reviewd same time as my team mate.
Like i commented, U did very well and it was good code refactoring as well

**Pull request merged**